CMakeLists.txt

```diff
@@ -1,15 +1,51 @@
 cmake_minimum_required(VERSION 3.10)
-project(recycle)
+
+project(recycle
+  VERSION 6.0.0
+  DESCRIPTION "Simple resource pool for recycling resources in C++"
+  LANGUAGES CXX)
 
 # Define library
 add_library(recycle INTERFACE)
 target_compile_features(recycle INTERFACE cxx_std_14)
-target_include_directories(recycle INTERFACE src/)
+target_include_directories(recycle INTERFACE $<BUILD_INTERFACE:src> $<INSTALL_INTERFACE:include>)
 add_library(steinwurf::recycle ALIAS recycle)
 
+include(GNUInstallDirs)
+set(RECYCLE_CONFIG_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/recycle" CACHE PATH "recycle config directory")
+
 # Install headers
 install(
   DIRECTORY ./src/recycle
-  DESTINATION ${CMAKE_INSTALL_PREFIX}/include
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
   FILES_MATCHING
   PATTERN *.hpp)
+
+# Install targets
+install(TARGETS recycle EXPORT recycle-targets)
+install(
+  EXPORT recycle-targets
+  DESTINATION ${RECYCLE_CONFIG_DIR}
+  NAMESPACE steinwurf::)
```

Comment:
I need help understanding the purpose of lines 24-29.

Reply:
This call will create the file `recycle-targets.cmake` in the dir `${RECYCLE_CONFIG_DIR}`.
The targets defined in this file will be prefixed with `steinwurf::`.

Downstream projects can then link these targets e.g.
```
find_package(recycle)
target_link_libraries(<TARGET> PUBLIC steinwurf::recycle)
```